src/strategies/zora-drops-filter-contracts/transformer.mjs

```diff
@@ -0,0 +1,92 @@
+// @format
+import { decodeLog } from "eth-fun";
+
+import logger from "../../logger.mjs";
+import { parseJSON } from "../../utils.mjs";
+
+export const name = "zora-drops-filter-contracts";
+const log = logger(name);
+export const version = "0.1.0";
+
+export function onClose() {
+  log("closed");
+  return {
+    write: null,
+    messages: [],
+  };
+}
+
+export function onError(error) {
+  log(error.toString());
+  throw error;
+}
+
+function decodeDropLog(log) {
+  const topics = log.topics;
+  topics.shift();
+  const result = decodeLog(
+    // index_topic_1 address creator, index_topic_2 address editionContractAddress, uint256 editionSize
+    [
+      {
+        type: "address",
+        name: "creator",
+        indexed: true,
+      },
+      {
+        type: "address",
+        name: "editionContractAddress",
+        indexed: true,
+      },
+      {
+        type: "uint256",
+        name: "editionSize",
+      },
+    ],
+    log.data,
+    topics
+  );
+
+  if (result) {
+    for (const k of ["creator", "editionContractAddress"]) {
+      result[k] = result?.[k].toLowerCase?.();
```

Comment:
What is going on here? Why can't we e.g. just get "creator" or "editionContractAddress"?

Reply:
This is done to ensure that the addresses are lowercase. So far as I can tell, from the event logs on etherscan they already are, but I do not think it is guaranteed to be the case. The same thing is done for soundxyz here: https://github.com/neume-network/strategies/blob/5b087b54660abba2a4c3570db5805c847c3700bb/src/strategies/soundxyz-filter-contracts/transformer.mjs#L46

The creator address is currently not included in the output.

Reply:
ah, ok but why are we iterating through creator and editionContractAddress?

Reply:
OK. Would unwinding the for loop resolve the issue for you? E.g.
```js
result.creator = result?.creator.toLowerCase?.();
result.editionContractAddress = result?.editionContractAddress.toLowerCase?.();
```

Reply:
oh sorry my bad, I didn't realize that we were storing result[k]. You can leave it as is.